packages/block-editor/src/components/inserter/block-list.js

```diff
@@ -0,0 +1,461 @@
+/**
+ * External dependencies
+ */
+import {
+	map,
+	pick,
+	includes,
+	filter,
+	findIndex,
+	flow,
+	sortBy,
+	groupBy,
+	isEmpty,
+	without,
+} from 'lodash';
+import scrollIntoView from 'dom-scroll-into-view';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x, _n, sprintf } from '@wordpress/i18n';
+import { PanelBody, withSpokenMessages } from '@wordpress/components';
+import { addQueryArgs } from '@wordpress/url';
+import { controlsRepeat } from '@wordpress/icons';
+import { speak } from '@wordpress/a11y';
+import { createBlock, isUnmodifiedDefaultBlock } from '@wordpress/blocks';
+import { useMemo, useEffect, useState, useRef } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { compose, withSafeTimeout } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import BlockTypesList from '../block-types-list';
+import ChildBlocks from './child-blocks';
+import __experimentalInserterMenuExtension from '../inserter-menu-extension';
+import { searchItems } from './search-items';
+
+// Copied over from the Columns block. It seems like it should become part of public API.
+const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
+	return map(
+		innerBlocksTemplate,
+		( [ name, attributes, innerBlocks = [] ] ) =>
+			createBlock(
+				name,
+				attributes,
+				createBlocksFromInnerBlocksTemplate( innerBlocks )
+			)
+	);
+};
+
+const getBlockNamespace = ( item ) => item.name.split( '/' )[ 0 ];
+
+const MAX_SUGGESTED_ITEMS = 9;
+
+function InserterBlockList( {
+	clientId,
+	isAppender,
+	rootClientId,
+	onSelect,
+	onHover,
+	__experimentalSelectBlockOnInsert: selectBlockOnInsert,
+	filterValue,
+	debouncedSpeak,
+	setTimeout: safeSetTimeout,
+} ) {
+	const {
+		categories,
+		collections,
+		items,
+		rootChildBlocks,
+		getSelectedBlock,
+		destinationRootClientId,
+		getBlockIndex,
+		getBlockSelectionEnd,
+		getBlockOrder,
+		fetchReusableBlocks,
+	} = useSelect(
+		( select ) => {
+			const {
+				getInserterItems,
+				getBlockName,
+				getBlockRootClientId,
+				getBlockSelectionEnd: _getBlockSelectionEnd,
+				getSettings,
+			} = select( 'core/block-editor' );
+			const {
+				getCategories,
+				getCollections,
+				getChildBlockNames,
+			} = select( 'core/blocks' );
+
+			let destRootClientId = rootClientId;
+			if ( ! destRootClientId && ! clientId && ! isAppender ) {
+				const end = _getBlockSelectionEnd();
+				if ( end ) {
+					destRootClientId = getBlockRootClientId( end ) || undefined;
+				}
+			}
+			const destinationRootBlockName = getBlockName( destRootClientId );
+
+			const { __experimentalFetchReusableBlocks } = getSettings();
+
+			return {
+				categories: getCategories(),
+				collections: getCollections(),
+				rootChildBlocks: getChildBlockNames( destinationRootBlockName ),
+				items: getInserterItems( destinationRootClientId ),
```

Comment:
Another occurrence to fix:
```suggestion
				items: getInserterItems( destRootClientId ),
```

You can run unit tests locally to ensure it works properly.

Reply:
Missed it, thanks.